File: Fibonachi a0a1a2a3a4a5/With array/Ruby/fib.rb
quantity = gets()
quantity = quantity.to_i
i = 2
i = i.to_i
fibonachi_array = Array[quantity+1]
fibonachi_array[0] = 0
fibonachi_array[1] = 1
result = "" + fibonachi_array[1].to_s
while (i<=quantity)
	fibonachi_array[i] = fibonachi_array[i-1] + fibonachi_array[i-2]
	result = result + fibonachi_array[i].to_s
	i = i + 1
end
puts(result);

File: Fibonachi a4a2a0a1a3a5/With array/Java/fib.java
import java.util.Scanner;
public class fib{
	public static void main(String[] args){
		Scanner scan = new Scanner(System.in);
		int quantity, i;
		i=2;
		quantity = scan.nextInt();
		int[] fibonachiArray;
		fibonachiArray = new int[quantity+1];
		fibonachiArray[0]=0;
		fibonachiArray[1]=1;
		String result = "" + fibonachiArray[1];
		while (i<=quantity){
			fibonachiArray[i] = fibonachiArray[i-1] + fibonachiArray[i-2];
			if ((i/2)*2 == i){
				result = result + fibonachiArray[i];
			} else {
				result = fibonachiArray[i] + result;
			}
			i=i+1;
		} 

System.out.println(result);
	}
}